File: cses/cses1635.py
MOD = 10**9 + 7

n, m = map(int, input().split())
moedas = list(map(int, input().split()))

dp = [0] * (m+1)

dp[0] = 1

for i in range(1, m+1):
    for moeda in moedas:
        if i >= moeda:
            dp[i] = (dp[i] + dp[i-moeda]) % MOD

print(dp[m])

File: cses/cses1635.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int MOD = 1e9 + 7;
    int n, m; cin >> n >> m;
    vector<int> moedas(n);
    for (int i = 0; i < n; i++){
        int n; cin >> n;
        moedas[i] = n;
    }
    vector<int> dp(m+1);
    dp[0] = 1;
    for (int i = 1; i<= m; i++){
        for (int moeda: moedas){
            if (i >= moeda){
                dp[i] = (dp[i] + dp[i-moeda]) % MOD;
            }
        }
    }
    cout << dp[m] << "\n";
    return 0;
}